tests/Hedgehog.Tests/GenTests.fs

```diff
@@ -1,94 +1,76 @@
-﻿module Hedgehog.Tests.GenTests
+module Hedgehog.Tests.GenTests
 
 open Hedgehog
 open Hedgehog.Gen.Operators
-open Swensen.Unquote
-open Xunit
+open TestHelpers
 
-[<Theory>]
-[<InlineData(  8)>]
-[<InlineData( 16)>]
-[<InlineData( 32)>]
-[<InlineData( 64)>]
-[<InlineData(128)>]
-[<InlineData(256)>]
-[<InlineData(512)>]
-let ``dateTime creates System.DateTime instances`` count =
-    let actual = Gen.dateTime (Range.constant System.DateTime.MinValue System.DateTime.MaxValue) |> Gen.sample 0 count
-    actual
-    |> List.distinct
-    |> List.length
-    =! actual.Length
+let dtRange = Range.constantFrom (System.DateTime (2000, 1, 1)) System.DateTime.MinValue System.DateTime.MaxValue
```

Comment:
Can we inline this (as it currently is in `master`) instead of extracting it to `dtRange`? It's nice to have autonomous tests, where each ([setup](http://xunitpatterns.com/fixture%20setup.html)) phase is just right there.

Reply:
I looked at all differences in tests from master and made the same changes on fable-test branch. The only difference in test bodies now is not using Unquote because it is not compatible with fable so instead of `test <@ some_expression @>` we now have `Expect.isTrue ( some_expression )`. Everything else should be completely the same. If you find some differences please tell me and I will correct it.

Reply:
One of the reasons we're using Unquote is because the test failure messages can be given as step-by-step F# expression evaluations. Dropping Unquote is a trade-off here, but I think it's worth it since we'll have one test-suite for both .NET and Fable.

Do you also agree @adam-becker and @TysonMN?

Reply:
@moodmosaic I mostly agree. I think having `Unquote` would be worthwhile, but I don't think it should prevent this PR being merged. I can't look at/modify the current tests because of an issue with ionide/fsac, so having these would be excellent for me.

Reply:
I also want just one test suit.  I am not convinced that the tradeoff must be so big.  What I say next doesn't have to be achieved in this PR.

> The only difference in test bodies now is not using Unquote because it is not compatible with fable so instead of `test <@ some_expression @>` we now have `Expect.isTrue ( some_expression )`.

It seems like it should be possible to me to write this as
```fsharp
let ourTest (expr: Expr<bool>) =
#if FABLE_COMPILER
    expr |> Expr.eval |> Expect.isTrue
#else
    expr |> test // this test is from Swensen.Unquote
#endif
```

where `Expr.eval` has type `Expr<'a> -> 'a`.  One possible implementation that works for my limited use case at work is
```fsharp
open Microsoft.FSharp.Quotations
open FSharp.Linq.RuntimeHelpers.LeafExpressionConverter
let eval (expr: Expr<'a>) =
  expr |> EvaluateQuotation :?> 'a
```

My understanding is that that implementation cannot handle all possible expressions.  In my notes, I wrote down how one might reach two different implementations.

> This evaluator can evaluate "simple" expressions.
 > It is contained in FSharp.Core.dll,
 > so no additional dependency is needed to use it.
 > 
 > FSharp.Quotations.Evaluator can evaluate more, but not all, expressions.
 > http://fsprojects.github.io/FSharp.Quotations.Evaluator/
 > 
 > Swensen.Unquote contains another evaluator that it uses internally.
 > https://github.com/SwensenSoftware/Unquote

Doesn't seem like the evaluator from Swensen.Unqoute is an option since lack for Fable by Swensen.Unquote is the whole issue.

Reply:
@TysonMN Completely agree that we shouldn't lose `Unquote`, but I think we should restore that in a follow up PR. I can make an issue for `0.10.0` to make sure we do that before releasing.